File: dotnet-backend/src/Application/DTOs/AuthDtos.cs
namespace Application.DTOs;

/// <summary>
/// Represents the request data required for a user to log in.
/// </summary>
/// <param name="Username">
/// The unique username of the user attempting to log in. 
/// This is normally used to identify and authenticate the user.
/// </param>
/// <param name="Password">
/// The plain-text password provided by the user for authentication. 
/// Should be securely transmitted (e.g., over HTTPS).
/// </param>
public record LoginRequest(string Username, string Password);

/// <summary>
/// Represents the response returned after a successful login operation,
/// including tokens and session information.
/// </summary>
/// <param name="AccessToken">
/// The JWT or similar token to be used for authorizing API requests.
/// </param>
/// <param name="RefreshToken">
/// The token to obtain a new access token after expiration, used to maintain user sessions.
/// </param>
/// <param name="ExpiresInMinutes">
/// The number of minutes until the access token expires.
/// </param>
/// <param name="Role">
/// The role or authorization level assigned to the authenticated user
/// (e.g., "User", "Admin").
/// </param>
public record LoginResponse(string AccessToken, string RefreshToken, int ExpiresInMinutes, string Role);

/// <summary>
/// Represents the request data required for registering a new user.
/// </summary>
/// <param name="Username">
/// The desired username for the new user. Must be unique in the system.
/// </param>
/// <param name="Email">
/// The email address of the user. Used for contact and account recovery.
/// </param>
/// <param name="Password">
/// The plain-text password for the user account. Should be stored securely (hashed).
/// </param>
public record RegisterRequest(string Username, string Email, string Password);

File: python-backend/src/application/dtos/auth_dtos.py
from pydantic import BaseModel, Field

# -----------------------------------------------------------------------------
# Data Transfer Objects (DTOs) for Authentication
#
# These classes define the structure for payloads sent to and received from the
# authentication endpoints (login, registration, and token refresh).
# They ensure consistent, well-documented API contracts for auth workflows.
# -----------------------------------------------------------------------------

class LoginRequest(BaseModel):
    """
    Request schema for user login.

    Attributes:
        username (str): The user's username for identification.
        password (str): The plaintext password submitted for authentication.
    """
    username: str  # Username credential for login
    password: str  # Plaintext password credential

class LoginResponse(BaseModel):
    """
    Response schema sent after a successful login.

    Attributes:
        access_token (str): JWT for short-term API access (returned as 'accessToken').
        refresh_token (str): Token allowing access token renewal (returned as 'refreshToken').
        expires_in_minutes (int): Minutes until the access token expires (alias 'expiresInMinutes').
        role (str): User's role (e.g., "admin", "user").
    """
    access_token: str = Field(serialization_alias="accessToken")  # JWT access token for requests
    refresh_token: str = Field(serialization_alias="refreshToken")  # Token to refresh access token
    expires_in_minutes: int = Field(serialization_alias="expiresInMinutes")  # Expiry duration
    role: str  # Role/privilege assigned to user

class RegisterRequest(BaseModel):
    """
    Request schema for registering a new user.

    Attributes:
        username (str): Chosen username.
        email (str): Valid user email address.
        password (str): Secure password.
    """
    username: str  # Desired username
    email: str     # User's email address
    password: str  # Desired password

class RefreshTokenRequest(BaseModel):
    """
    Request schema for refreshing an expired access token.

    Attributes:
        refresh_token (str): Refresh token used to obtain a new access token.
    """
    refresh_token: str = Field(alias="refreshToken")  # Provided refresh token for renewal
